### PR TITLE
use fstream instead of crudy old C buffer with vlas

### PR DESCRIPTION
**Pull request by fundies**

### CODE REVIEW

File: CompilerSource/gcc_interface/gcc_backend.h [hunk 1]
```diff
@@ -34,23 +34,14 @@
...
-    char a[sz + 1];
-    sz = fread(a, 1, sz, pt);
-    fclose(pt);
-
-    a[sz] = 0;
-    return a;
-  }
+  std::ifstream t(fn);
+  std::stringstream buffer;
+  buffer << t.rdbuf();
```
Comment (JoshDreamland) [inline CompilerSource/gcc_interface/gcc_backend.h]: What is the behavior of this line when `!t.is_open()`?

Comment (fundies) [reply CompilerSource/gcc_interface/gcc_backend.h]: it returns an empty string. Do you want to add a warning for that case?

Comment (JoshDreamland) [reply CompilerSource/gcc_interface/gcc_backend.h]: Not a strong preference, but I can't find any documentation that defines that behavior. I can take a look at the standard to verify it, I guess. Printing the warning and returning the empty string guarantees that we're fine, though, so I certainly am not opposed to that.

Comment (RobertBColton) [reply CompilerSource/gcc_interface/gcc_backend.h]: Does fc stand for fundie's cat?

Comment (fundies) [reply CompilerSource/gcc_interface/gcc_backend.h]: added warning for this case

**Comment by JoshDreamland** [comment]
Actually, hold that thought. I looked around to see whether `basic_ostream::operator<<(basic_streambuf*)` preallocated based on the size of the file, and I found no such guarantee. This method might be slower than the one we have. I don't know that it will be that much slower, but it might be worth investigating.

**Comment by fundies** [comment]
If you really care bout speed why aren't you using platform specific IO? This code is shorter and looks nicer than what we have. Also, if msvc can't implement a vla what makes you think its gonna implement FSEEK n etc...

**Comment by JoshDreamland** [comment]
The difference is that those are actually part of the standard. Anyway, I'll assume `sstream` does the right thing.

**Pull request merged**